conventions/cli.md

```diff
@@ -0,0 +1,56 @@
+# Guidelines for kn Commands
+
+Commands are generally of the form `kn <resource> <verb>`.
+
+## Resource
+
+The <group>.knative.dev Kind, singluar and lowercase. For example, `service` for
+`serving.knative.dev/service` or `trigger` for `eventing.knative.dev/trigger`.
+
+## Verb
+
+If the thing the user's doing can be described by the following commands, these
+should be the name of the verb:
+
+* `describe` prints detailed information about a single resource. It can include
+  status information of related or child resources, too.
+
+* `list` prints summary information about all resources of a type, possibly
+  filtered by parent or label selector.
+
+* `create` creates a resource. Accepts a `--force` flag to create-or-replace.
+
+* `update` updates a resource based on the changes the user would like to make.
+
+* `delete` deletes a resource
+
+For a given resource there should be parallelism between arguments to `--create`
+and `--update` as much as possible.
+
+Other domain-specific verbs are possible, like `set-traffic` for a Knative
+Serving Service.
+
+## Arguments
+
+### Positionals
+
+Where there's a single target resource, it should be a positional argument.
```

Comment:
Maybe add that this positional always reference an entity of <noun>,
e.g. for a `kn revision list <something>` , `<something>` is always the name of a revision (and not e.g. a service)

Reply:
Thanks